src/nxt_otel.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) F5, Inc.
+ */
+
+#ifndef _NXT_OTEL_H_INCLUDED_
+#define _NXT_OTEL_H_INCLUDED_
+
+#include <nxt_router.h>
+#include <nxt_string.h>
+
+// forward declared
+struct nxt_http_field_t;
+struct nxt_conf_validation_t;
+struct nxt_conf_value_t;
+struct nxt_http_request_t;
```

Comment:
@ac000 what exactly bugs you on this? I know you talked about xmas tree ordering of variables and it is technically in the coding style guide in the confluence page, but the existing code very very loosely follows this (see https://github.com/nginx/unit/blob/master/src/nxt_java.c#L147-L150, https://github.com/nginx/unit/blob/master/src/nxt_listen_socket.c#L33-L45, https://github.com/nginx/unit/blob/master/src/nxt_conn_accept.c#L21-L34 as 3 very quick examples I found when clicking on random files).

I'm not against making this a rule, **provided** we can make this an **automated** rule. `clang-format` doesn't have a directive for this, nor does indent, nor does uncrustify.

Do you know of a tool we can wire up to github actions that would yell at us about the order?

Reply:
Seems GH is being wonky and this comment wasn't showing up, I'll stick my other reply here...

> @ac000 what exactly bugs you on this?

It's the fact we need the forward declarations.

IIRC this due to errors from including the required header files. I
would be interested in knowing about the specific issue you were having.

I had a similar issue in the cgroup patches and I needed to forward
declare a struct in the nxt_cgroup.h file.

However upon review from Alex it was determined that a better solution
was to move the thing that was requiring the forward declaration to a
more appropriate place and thus there was no need for it.

Reply:
Not that this had anything to do with variable ordering... the whole codingstyle thing seems to have been a red herring...

But to answer your points about the examples you found (and I'm no fan of the way we do it...)

> <https://github.com/nginx/unit/blob/master/src/nxt_java.c#L147-L150>

Looks right to me.

> <https://github.com/nginx/unit/blob/master/src/nxt_listen_socket.c#L33-L45>

Also looks right, taking into the account the conditional.

> <https://github.com/nginx/unit/blob/master/src/nxt_conn_accept.c#L21-L34>

Is something different entirely...

Reply:
Here's the relevant (albeit brief) [discussion](https://github.com/nginx/unit/pull/734#discussion_r944733700) of the cgroup issue.

Reply:
Looks like those forward declarations aren't even needed. I.e. It still builds for me with this patch

```diff
diff --git ./src/nxt_otel.h ./src/nxt_otel.h
index f833b1ee..1bd34d06 100644
--- ./src/nxt_otel.h
+++ ./src/nxt_otel.h
@@ -8,11 +8,6 @@
 #include <nxt_router.h>
 #include <nxt_string.h>
 
-// forward declared
-struct nxt_http_field_t;
-struct nxt_conf_validation_t;
-struct nxt_conf_value_t;
-struct nxt_http_request_t;
 
 extern void     nxt_otel_rs_send_trace(void *trace);
 extern void   * nxt_otel_rs_get_or_create_trace(u_char *trace_id);
```

Reply:
Its all fun and games until in the future someone includes nxt_otel.h without including the headers that define those forward declared types. 

Then the include breaks and nxt_otel.h needs to be fixed in an otherwise unrelated merge.

Reply:
Well, if that ever did become an issue we'd fix it.

As already mentioned our header situation is less than ideal, however, between `nxt_router.h` and `nxt_string.h` they include everything that's required.